src/Microsoft.ML.Data/TrainCatalog.cs

```diff
@@ -61,14 +85,68 @@ public abstract class TrainCatalogBase
                 Complement = false
             }, data);
 
-            return (trainFilter, testFilter);
+            return new TrainTestData(trainFilter, testFilter);
+        }
+
+        /// <summary>
+        /// Results for specific cross validation fold.
+        /// </summary>
+        protected internal struct CrossValidationResult
+        {
+            /// <summary>
+            /// Model trained during cross validation fold.
+            /// </summary>
+            public readonly ITransformer Model;
+            /// <summary>
+            /// <see cref="IDataView"/> for scored fold.
+            /// </summary>
+            public readonly IDataView Scores;
+            /// <summary>
+            /// Fold number.
+            /// </summary>
+            public readonly int Fold;
+
+            public CrossValidationResult(ITransformer model, IDataView scores, int fold)
+            {
+                Model = model;
+                Scores = scores;
+                Fold = fold;
+            }
+        }
+
+        public class CrossValidationResult<T> where T : class
+        {
+            /// <summary>
+            /// Metrics for cross validation fold.
+            /// </summary>
+            public readonly T Metrics;
+            /// <summary>
+            /// Model trained during cross validation fold.
+            /// </summary>
+            public readonly ITransformer Model;
+            /// <summary>
+            /// <see cref="IDataView"/> for scored fold.
```

Comment:
>for scored fold [](start = 40, length = 15)

Scored test fold? Any one has good ideas?

Reply:
"The scored hold-out set for this fold"

---
In reply to: [255714995](https://github.com/dotnet/machinelearning/pull/2507#discussion_r255714995) [](ancestors = 255714995)

Reply:
`ScoredHoldOutSet` is literal, but long.

---
In reply to: [255726140](https://github.com/dotnet/machinelearning/pull/2507#discussion_r255726140) [](ancestors = 255726140,255714995)